### ISSUE TITLE
Core suddenly rebooting - X-Men vs. Street Fighter

### ISSUE THREAD

**Issue by luciusbono** [jotego/jtcps#125]
I'm trying to chase down an intermittent issue with X-Men vs. Street Fighter where the core will suddenly reboot. It doesn't happen every session, but does happen _most_ sessions around 3 rounds into the game. 

I recognize this is super vague, but I'm new to the MiSTer platform and haven't been able to locate any logs that may be helpful - and it's not consistent enough where I've been able to find any specific trigger conditions. Only the core reboots (screen freezes and the core boot sequence occurs). I have only noticed this on X-Men vs. Street Fighter but it's the ROM/core I play the most. 

Running the latest version of the core / MiSTer from `update_all`. MiSTer is using a high quality power supply / is well cooled (it's using the MiSTer Multisystem daughterboard). Input device is an 8BitDo Arcade Stick. 

Any guidance you could provide to help produce a better problem report would be greatly appreciated - certainly not expecting answers with this vague of a problem report!

**Comment by jotego** [comment]
Thank you for your kind report.
This could be related to #110. Can you try playing with the Phoenix ROM version? That version doesn't use encryption and if worked, it would point to the problem.

**Comment by luciusbono** [comment]
Sorry for the long delay, I was traveling last week and didn't have enough time to test the Phoenix ROM. 
Testing with the USA Phoenix ROM for about two hours has yielded _zero_ resets so I suspect your hunch about encryption being the problem is correct. 

Is there any other testing I can do to help?

**Comment by jotego** [comment]
Thank you very much. This is precious information. I'll try to spend some time on it this week.

**Comment by venalaine** [comment]
I can also confirm this issue. I have been playing this game during past couple of days and sometimes the game just reboots. It seems that the issue is not related to certain level or anything. Sometimes it happens after 5 min of playing, sometimes after 30 minutes or something.

**Comment by ibrahima-whatever** [comment]
Hi,

This happens fairly regularly in single player. Steps to reproduce below:

1. Pick a team that includes Akuma / Gouki. He can be selected by moving up from Magneto or another character on the top row.

2. In a match, use Akuma’s super; Raging Demon / Shun Goku Satsu against the CPU when one of their characters is on relatively low life (eg less than a third or so). This super can be used by pressing ‘jab, jab, direction forward, short, fierce’. If you are successful Akuma will trigger a super animation where he slowly floats towards the opponent, then grabs them with a white screen flash. Another description for the input depending on preference is ‘light punch, light punch, forward, light kick, heavy punch’. You may need to practice this move to get used to the input

3.  If you have been successful the following things should happen:
* Akuma defeats the CPU opponent with the super
* The CPU tag partner character jumps onto the screen after the white flash and the super combo has completed
* The game resets when the CPU tag partner attempts to attack

Please note that this happens between 50-70% of the time when playing against the CPU. In some cases it is possible that the game will not reset even when you have input the super correctly and defeated the first CPU opponent.

I have noticed that a similar crash also occurs in Marvel Super Heroes using the hidden character Anita when playing against a CPU controlled Shuma Gorath on their stage.

If needed I can do a recording of the Akuma crash. Please let me know if that would be useful.

**Comment by ibrahima-whatever** [comment]
Just noted re: Phoenix set. I will try and replicate it on that set over today and let you know how I get on.

**Comment by jotego** [comment]
Thank you. Please let us know if the Phoenix set is ok. I tried a few things about this on the related issue #110 but I couldn't fix it. I will try a different approach this week to find the root cause.

**Comment by ibrahima-whatever** [comment]
Retested on the Phoenix USA 961004 rom. No crash. Played through arcade mode and landed the super roughly five or six times to KO cpu opponent. Game continued as normal.

Out of interest I retested on the standard USA 961004 rom. No crash. Tried to reproduce several times.

Retested on the Euro 961004 rom. Crash. This was reproduced several times.

Retested on the Euro 960910 rom. No crash.

So at this point can’t guarantee that the Phoenix rom includes a change that prevents this particular crash on Xmen Vs Street Fighter when ko’ing a CPU player with Raging Demon, though I understand that this has prevented crashes on other romsets in similar situation (eg the old deejay crash on ST).

I would assume that outside of the region code and related changes the two 961004 romsets are identical, but again I could understand if that isn’t the case for whatever reason (eg change in how the cpu controlled character behaves, or changes to game logic or similar).

In the interest of narrowing this down might be worthwhile to note that I typically play with in-game turbo setting enabled (eg ‘normal’ or ‘turbo’ on character select screen). I usually pick Akuma as first character but have also seen this happen when he is secondary. I have been able to reproduce this when he is partnered with Ryu or Magneto, so again tag partner doesn’t seem to impact this.
